vector/src/test/java/im/vector/app/features/poll/create/CreatePollViewModelTest.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.poll.create
+
+import com.airbnb.mvrx.test.MvRxTestRule
+import im.vector.app.features.poll.PollMode
+import im.vector.app.features.poll.create.CreatePollViewStates.createPollArgs
+import im.vector.app.features.poll.create.CreatePollViewStates.editPollArgs
+import im.vector.app.features.poll.create.CreatePollViewStates.fakeOptions
+import im.vector.app.features.poll.create.CreatePollViewStates.fakeQuestion
+import im.vector.app.features.poll.create.CreatePollViewStates.fakeRoomId
+import im.vector.app.features.poll.create.CreatePollViewStates.initialCreatePollViewState
+import im.vector.app.features.poll.create.CreatePollViewStates.pollViewStateWithOnlyQuestion
+import im.vector.app.features.poll.create.CreatePollViewStates.pollViewStateWithQuestionAndEnoughOptions
+import im.vector.app.features.poll.create.CreatePollViewStates.pollViewStateWithQuestionAndMaxOptions
+import im.vector.app.features.poll.create.CreatePollViewStates.pollViewStateWithQuestionAndNotEnoughOptions
+import im.vector.app.features.poll.create.CreatePollViewStates.pollViewStateWithoutQuestionAndEnoughOptions
+import im.vector.app.test.test
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.shouldBe
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.session.room.Room
+import org.matrix.android.sdk.api.session.room.model.message.PollType
+import org.matrix.android.sdk.api.session.room.send.SendService
+import org.matrix.android.sdk.api.util.Cancelable
+
+class CreatePollViewModelTest {
+
+    @get:Rule
+    val mvrxTestRule = MvRxTestRule()
+
+    private val fakeSession = mockk<Session>()
```

Comment:
We should use/create FakeObject classes to be reusable for other unit tests. Here you can use `FakeSession` and `FakeRoom`. You can add `givenXXX` methods in them to mock some calls (see existing examples). For the `SendService`, a `FakeSendService` can be created and instantiated inside the `FakeRoom` class. We may only use `mockk<>` inside tests class when it is very specific to the tested class or simple object.

Reply:
Right, done.